architecture/core.rst

```diff
@@ -390,6 +390,149 @@ normal world supports scheduling of processes. Even on UP systems, supporting
+Interrupt consumers main API functions are:
+
+    - ``interrupt_enable()`` and ``interrupt_disable()`` to respectively
+      enable or disable an interrupt.
+
+    - ``interrupt_mask()`` and ``interrupt_unmask()`` to respectively mask
+or unmask an interrupt. Masking of an enabled interrupt temporarily disables
+the interrupt while unmasking enables a previously masked interrupt.
+``interrupt_mask()`` and ``interrupt_unmask()`` are allowed to be called
+from an interrupt context, but ``interrupt_enable()`` and
+``interrupt_disable()`` not so.
+
+    - ``interrupt_configure()`` to configure an interrupt detection mode
+      and priority.
+
+    - ``interrupt_add_handler()`` to configure and interrupt and register
+      a interrupt handler function, see below.
+
+    - ``interrupt_remove_handler()`` to unregister an interrupt handler
+      function from an interrupt.
+
+**Interrupt controller drivers**
+
+An interrupt controller instance, named chip (``struct itr_chip``) defines
+operation function handlers for management of the interrupt(s) it controls.
+An interrupt chip driver must provide operation handler functions ``.add``,
+``.mask``, ``.unmask``, ``.enable`` and ``.disable``. There are other
+operation handler functions that are optional, as ``.rasie_pi``,
+``.raise_sgi`` and ``.set_priority``.
+
+An interrupt chip driver registers the controller instance with API
+function ``itr_chip_init()``. The driver calls the registered interrupt
+consumer(s) handler(s) with API function ``interrupt_call_handlers()``.
+
+**CPU main interrupt controller driver**
+
+CPU main interrupt controller (e.g. a GIC instance on Arm architecture CPUs)
+driver must register itself as main controller using API function
+``interrupt_main_init()``. The function is in charge of calling
+``itr_chip_init()`` for that chip instance.
+
+Interrupt consumer drivers get can get the CPU main interrupt controller
+reference with API function ``interrupt_get_main_chip()``.
+
+**Interrupt handlers**
+
+Interrupt handler functions are callback functions registered by interrupt
+consumer drivers that core shall call when the related interrupt occurs.
+Structure ``struct itr_handler`` references a handler. It contains the
+handler function entry point, the interrupt number, the interrupt controller
+device and a few more parameters.
+
+An interrupt handler function return value is of type ``enum itr_return``.
+It shall return ``ITRR_HANDLED`` when the interrupt is served and
+``ITRR_NONE`` when the interrupt cannot be served.
+
+The interrupt handler runs in an interrupt context rather than a thread
+context. When this occurs, all other interrupts are masked, necessitating fast
+execution of the interrupt handler to avoid delaying or missing out on other
+interrupts. If there are long lasting operations to do on that interrupt
+occurrence, the interrupt handler should request execution of the OP-TEE
+bottom half thread, see :ref:`notifications`.
```

Comment:
Yet another suggestion:

s/If there ... bottom half thread/When an interrupt occurs that requires the completion of long-running operations, the interrupt handler should request that the OP-TEE bottom half thread be executed./

Reply:
Ok, I'll reword the end:

_When an interrupt occurs that requires the completion of long-running operations, the interrupt handler should request ~~that~~ the OP-TEE bottom half thread **to execute those operations**, see ..._